### PR TITLE
Deprecation migration part 1

### PR DESCRIPTION
**Pull request by data-ux**
Sandbox.getAjaxUrl() changed to Oskari.urls.getRoute()

### CODE REVIEW

File: bundles/admin/admin/DefaultViews.js [hunk 1]
```diff
@@ -53,16 +53,15 @@ Oskari.clazz.define('Oskari.admin.bundle.admin.DefaultViews', function(locale, p
...
+        var me = this;
 
         jQuery.ajax({
             type: 'GET',
             dataType: 'json',
             data: {
                 action_route: 'SystemViews'
             },
-            url: sb.getAjaxUrl(),
+            url: Oskari.urls.getRoute(),
```
Comment (ZakarFin) [inline bundles/admin/admin/DefaultViews.js]: Change this to Oskari.urls.getRoute('SystemViews') and take out the data-block.

File: bundles/admin/admin/DefaultViews.js [hunk 2]
```diff
@@ -108,7 +107,7 @@ Oskari.clazz.define('Oskari.admin.bundle.admin.DefaultViews', function(locale, p
             type: 'POST',
             dataType: 'json',
             data: data,
-            url: sb.getAjaxUrl(),
+            url: Oskari.urls.getRoute(),
```
Comment (ZakarFin) [inline bundles/admin/admin/DefaultViews.js]: Same here: give 'SystemViews' to getRoute() and take action_route: 'SystemViews' out of the payload.

File: bundles/admin/metrics/instance.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ Oskari.clazz.define("Oskari.admin.bundle.metrics.MetricsAdminBundleInstance",
             jQuery.ajax({
                 dataType : "json",
                 type : "GET",
-                url : sandbox.getAjaxUrl(),
+                url : Oskari.urls.getRoute(),
```
Comment (ZakarFin) [inline bundles/admin/metrics/instance.js]: Give 'Metrics' as param and take out data-block.

**Comment by data-ux** [comment]
Good catch! I additionally fixed some similar occurrences.

**Comment by ZakarFin** [comment]
Nice :+1:  There's still a few left that could be changed :)

File: bundles/framework/admin-users/AdminRoles.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ Oskari.clazz.define(
         this.instance = parent;
         this._localization = localization;
         this.sandbox = parent.getSandbox();
-        this.ajaxUrl = this.sandbox.getAjaxUrl();
+        this.ajaxUrl = Oskari.urls.getRoute();
```
Comment (ZakarFin) [inline bundles/framework/admin-users/AdminRoles.js]: Looks like you can add 'ManageRoles' as param and modify the usage by removing action_route=ManageRoles.

File: bundles/framework/divmanazer/component/FilterDialog.js [hunk 5]
```diff
@@ -896,7 +896,7 @@ Oskari.clazz.define('Oskari.userinterface.component.FilterDialog',
          */
         _loadWFSLayerPropertiesAndTypes:function (layer_id, prevJson, cb, clickedFeatures, selectedTemporaryFeatures) {
             var me = this,
-                url = me.sandbox.getAjaxUrl()
+                url = Oskari.urls.getRoute()
```
Comment (ZakarFin) [inline bundles/framework/divmanazer/component/FilterDialog.js]: Looks like url isn't used here.

File: bundles/framework/download-basket/Basket.js [hunk 6]
```diff
@@ -164,7 +164,7 @@ Oskari.clazz.define(
             };
             var strUserDetails = JSON.stringify(userDetails);
 
-            var ajaxUrl = me._sandbox.getAjaxUrl();
+            var ajaxUrl = Oskari.urls.getRoute();
```
Comment (ZakarFin) [inline bundles/framework/download-basket/Basket.js]: You could add 'DownloadInfo' as param here and remove it from the concatenated url below.

File: bundles/framework/myplaces2/service/MyPlacesService.js [hunk 7]
```diff
@@ -632,7 +632,7 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces2.service.MyPlacesServic
                 callback(false);
                 return;
             }
-            var ajaxUrl = me._sandbox.getAjaxUrl();
+            var ajaxUrl = Oskari.urls.getRoute();
```
Comment (ZakarFin) [inline bundles/framework/myplaces2/service/MyPlacesService.js]: You could add 'PublishMyPlaceLayer' as param here.

File: bundles/framework/publisher2/view/PublisherSidebar.js [hunk 8]
```diff
@@ -459,7 +459,7 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PublisherSidebar
         _publishMap: function (selections) {
             var me = this,
                 sandbox = me.instance.getSandbox(),
-                url = sandbox.getAjaxUrl(),
+                url = Oskari.urls.getRoute(),
```
Comment (ZakarFin) [inline bundles/framework/publisher2/view/PublisherSidebar.js]: You could add 'AppSetup' as param here.

File: bundles/integration/admin-layerselector/views/adminLayerSettingsView.js [hunk 9]
```diff
@@ -1427,7 +1427,7 @@ define([
             _setupSldStyles: function() {
                 var me = this,
                     elem = me.$el,
-                    baseUrl = me.options.instance.getSandbox().getAjaxUrl();
+                    baseUrl = Oskari.urls.getRoute();
```
Comment (ZakarFin) [inline bundles/integration/admin-layerselector/views/adminLayerSettingsView.js]: You could add 'SldStyles' as param here.

File: bundles/integration/admin-layerselector/views/adminLayerSettingsView.js [hunk 10]
```diff
@@ -1456,7 +1456,7 @@ define([
              */
             _saveSldStyle: function(sldName, sldXml) {
                 var me = this,
-                    baseUrl = me.options.instance.getSandbox().getAjaxUrl();
+                    baseUrl = Oskari.urls.getRoute();
```
Comment (ZakarFin) [inline bundles/integration/admin-layerselector/views/adminLayerSettingsView.js]: You could add 'SldStyles' as param here.

File: bundles/mapping/mapmodule/plugin/getinfo/GetInfoPlugin.js [hunk 11]
```diff
@@ -267,7 +267,7 @@ Oskari.clazz.define(
             var me = this,
                 dteMs = (new Date()).getTime(),
                 layerIds = me._buildLayerIdList(layers || this.getSandbox().findAllSelectedMapLayers()),
-                ajaxUrl = this.getSandbox().getAjaxUrl(),
+                ajaxUrl = Oskari.urls.getRoute(),
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/plugin/getinfo/GetInfoPlugin.js]: You could add 'GetFeatureInfoWMS' as param here.

**Comment by data-ux** [comment]
Fixed more.

File: bundles/framework/routingService/instance.js [hunk 12]
```diff
@@ -189,7 +189,7 @@ function () {
      */
     getRoute: function (params) {
         var me = this;
-            getRouteUrl = this.sandbox.getAjaxUrl('Routing');
+            getRouteUrl = Oskari.urls.getRoute('Routing');
```
Comment (ZakarFin) [inline bundles/framework/routingService/instance.js]: var missing here.

**Pull request merged**